include/dnscpp/tlsa.h

```diff
@@ -0,0 +1,170 @@
+        // this CA, but the entire chain to a trusted root-CA must still be
+        // valid.
+        certificate_authority_constraint = 0,
+
+        // The certificate used must match the TLSA record exactly, and it must
+        // also pass PKIX certification path validation to a trusted root-CA.
+        service_certificate_constraint = 1,
+
+        // The certificate used has a valid certification path pointing back to
+        // the certificate mention in this record, but there is no need for it
+        // to pass the PKIX certification path validation to a trusted root-CA.
+        trust_anchor_assertion = 2,
+
+        // The services uses a self-signed certificate. It is not signed by
+        // anyone else, and is exactly this record.
+        domain_issued_certificate = 3
+    };
+
+    /**
+     *  When connecting to the service and a certificate is received, the
+     *  selector field specifies which parts of it should be checked.
+     */
+    enum class Selector : uint8_t
+    {
+        // Select the entire certificate for matching.
+        entire_certificate = 0,
+
+        // Select just the public key for certificate matching. Matching the
+        // public key is often sufficient, as this is likely to be unique.
+        only_public_key = 1
+    };
+
+    /**
+     *  Matching type
+     */
+    enum class MatchingType : uint8_t
+    {
+        // The entire information selected is present in the certificate
+        // association data.
+        info_is_in_certificate_association_data = 0,
+
+        // Do a SHA-256 hash of the selected data.
+        do_SHA256 = 1,
+
+        // Do a SHA-512 hash of the selected data.
+        do_SHA512 = 2
+    };
+
+private:
+    /**
+     *  The certificate usage
+     *  @var CertificateUsage
+     */
+    CertificateUsage _usage;
+
+    /**
+     *  The selector
+     *  @var Selector
+     */
+    Selector _selector;
+
+    /**
+     *  The matching type
+     *  @var MatchingType
+     */
+    MatchingType _matching;
+
+    /**
+     *  The association data
+     *  @var unsigned char *
+     */
+    const unsigned char *_association;
+
+public:
+    /**
+     *  The constructor
+     *  @param  response        the response from which the record was extracted
+     *  @param  record          the record holding the CNAME
+     *  @throws std::runtime_error
+     */
+    TLSA(const Response &response, const Record &record) :
+        Extractor(record, TYPE_TLSA, 3),
+        _usage(static_cast<CertificateUsage>(record.data()[0])),
+        _selector(static_cast<Selector>(record.data()[1])),
+        _matching(static_cast<MatchingType>(record.data()[2])),
+        _association(record.data() + 3) {}
+
+    /**
+     *  Destructor
+     */
+    virtual ~TLSA() = default;
+
+    /**
+     *  How to verify the certificate?
+     *  @return the certificate usage
+     */
+    CertificateUsage certificateUsage() const { return _usage; }
```

Comment:
we don't need these members stored, as they're already in _record, simply get the correct byte and cast in the getters, like the rest of the library.